airlock/permissions.py

```diff
@@ -225,19 +225,20 @@ def user_can_update_file_on_request(
 
 
 def check_user_can_withdraw_file_from_request(
-    user: User, request: "ReleaseRequest", relpath: UrlPath
+    user: User, request: "ReleaseRequest", workspace: "Workspace", relpath: UrlPath
```

Comment:
does this need a sanity check that the request & workspace match?

Reply:
we don't do this anywhere else, I didn't previously mind

Reply:
We could check at least that the `ReleaseRequest.workspace` and `Workspace.name` match. I'm not sure we can check much more than that (current request on the workspace isn't necessarily the same as the release request passed in here, and if it isn't, it should hit the permission error as expected).

We could of course just pass the request and get the workspace based on the workspace name from the release request, and the user. But that means every permission check will redo that, when it's generally already been done in the view. And we'd have to call out to the Workspace model, which would introduce a circular import that we can't circumvent like we do with the typing imports.